src/components/core/ApplicationNavigationBar.tsx

```diff
@@ -1,25 +1,121 @@
-import { Navbar, Text } from "@mantine/core";
+import { Navbar, List } from "@mantine/core";
+import Link from "next/link";
+import React, { useState } from "react";
+import { FaChevronDown } from "react-icons/fa";
+
+interface NavigationItem {
+  title: string;
+  href: string;
+  subitems?: NavigationItem[];
+}
 
 interface ApplicationNavigationBarProps {
-	opened: boolean;
+  opened: boolean;
+  onClose: () => void;
 }
 
-export default function ApplicationNavigationBar({ opened }: ApplicationNavigationBarProps) {
-	return (
-		<Navbar
-			p="md"
-			hiddenBreakpoint={Number.MAX_VALUE} // XD
-			hidden={!opened}
-			width={
-				opened
-					? { sm: 200, lg: 300 }
-					: {
-							sm: 0,
-							lg: 0,
-					  }
-			}
-		>
-			<Text>Application navbar</Text>
-		</Navbar>
-	);
+export default function ApplicationNavigationBar({
+  opened,
+}: ApplicationNavigationBarProps) {
+  const [activeIndex, setActiveIndex] = useState<number | null>(null);
+
+  const navigationItems: NavigationItem[] = [
+    {
+      title: "Home",
+      href: "/",
+    },
+    {
+      title: "Undergraduate",
+      href: "/undergraduate",
+      subitems: [
+        {
+          title: "idk",
+          href: "/xd",
+        },
+      ],
+    },
+    {
+      title: "Study Plans",
+      href: "/study_plans",
+    },
+    {
+      title: "More",
+      href: "/more",
+    },
+  ];
+
+  const linkStyle = {
+    textDecoration: "none",
+    color: "inherit",
+    listStyle: "none",
+    padding: 0,
+    margin: 0,
+  };
+
+  const handleItemClick = (index: number) => {
+    if (index === activeIndex) {
+      setActiveIndex(null); // Toggle off if already active
+    } else {
+      setActiveIndex(index); // Activate index and show subitems
+    }
+  };
+
+  
+
+  return (
+    <Navbar
+      p="md"
+      hiddenBreakpoint={Number.MAX_VALUE}
+      hidden={!opened}
+      width={
+        opened
+          ? { sm: 200, lg: 300 }
+          : {
+              sm: 0,
+              lg: 0,
+            }
+      }
+    >
+      <div style={{ display: "flex", alignItems: "flex-start" }}>
```

Comment:
I'm assuming you have to hardcode the style like this because it can't be done with tailwind css right? idk XD @Lxkas  second opinion pls

Reply:
actually i tried the other ways but couldn't do it in an efficient way. that's why i need comments xD

Reply:
ahh oki oki Mister @Lxkas come here right now 🤣

cc: if you guys wanna see css styling:
@Naris167 @Ayham1590

Reply:
yes xD, pls help out this lost boy.